### PR TITLE
feat KHP3-7090: added  Status Filtering for Deceased Patients Based on Morgue Visit and Discharge Encounter

### PR DESCRIPTION
**Pull request by its-kios09**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

what I have is the following:-
- This feature will enable filtering of deceased patients based on their visit status in the morgue: i.e, admitted, awaiting admission, and discharged.
- Redesigning of the Mortuary view on patient chart

## Screenshots
morgueVisit-updated

## Related Issue
https://thepalladiumgroup.atlassian.net/browse/KHP3-7090
Jira
 
*None.*

## Other

*None.*

### CODE REVIEW

**Comment by ojwanganto** [comment]
Thanks for the work. Is it possible to configure the compartments to be locations under a location tagged as morgue? Something similar to beds in a ward?

File: packages/esm-morgue-app/src/types/index.ts [hunk 1]
```diff
@@ -30,12 +30,13 @@ export interface Patient {
     } | null;
   };
 }
-export interface DeceasedPatientResponse {
-  results: Patient[];
-}
+// export interface DeceasedPatientResponse {
+//   results: Patient[];
+// }
```
Comment (makombe) [inline packages/esm-morgue-app/src/types/index.ts:35]: You can remove this if it is no-longer needed

File: packages/esm-patient-clinical-view-app/src/deceased-panel/mortuary-summary/mortuary-summary.component.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  const patientUuid = getPatientUuidFromUrl();
+  const { person, isLoading } = usePerson(patientUuid);
+  const { activeVisit, isLoading: isActiveLoading } = useActiveMorgueVisit(patientUuid);
+  const startDate = activeVisit?.startDatetime;
+  if (isLoading || isActiveLoading) {
+    return (
+      <InlineLoading
+        status="active"
+        iconDescription="Loading"
+        description={t('loadData', 'Loading summary data.........')}
```
Comment (makombe) [inline packages/esm-patient-clinical-view-app/src/deceased-panel/mortuary-summary/mortuary-summary.component.tsx:30]:
```suggestion
        description={t('loadData', 'Loading summary data...')}
```

File: packages/esm-patient-clinical-view-app/src/deceased-panel/mortuary-summary/mortuary-summary.component.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    <div className={styles.summaryContainer}>
+      <p className={styles.morgueLabel}>{''}</p>
+      <div className={styles.metricList}>
+        <div className={styles.metrics}>
+          <div className={styles.wrapMetrics}>
+            <span className={styles.metricLabel}>{t('dateOfAdmission', 'Date of admission')}</span>
+            <span className={styles.metricValue}>{formatDateTime(startDate)}</span>
+          </div>
+          <div className={styles.wrapMetrics}>
+            <span className={styles.metricLabel}>{t('dateandTimeofDeath', 'Date and time of death')}</span>
```
Comment (makombe) [inline packages/esm-patient-clinical-view-app/src/deceased-panel/mortuary-summary/mortuary-summary.component.tsx:45]:
```suggestion
            <span className={styles.metricLabel}>{t('dateAndTimeofDeath', 'Date and time of death')}</span>
```

**Review by makombe — DISMISSED** [review]
Thanks @its-kios09 LGTM

**Comment by its-kios09** [comment]
> Thanks for the work. Is it possible to configure the compartments to be locations under a location tagged as morgue? Something similar to beds in a ward?

@ojwanganto Currently, I have just created a tag named compartment, in which I have tagged two existing locations but I am open to suggestions

**Comment by its-kios09** [comment]
@makombe Thanks let me updated the PR

**Comment by its-kios09** [comment]
@makombe You are welcomed i have updated the PR

**Review by makombe — APPROVED** [review]
Thanks @its-kios09

**Pull request merged**